rust/connlib/shared/src/proptest.rs

```diff
@@ -0,0 +1,42 @@
+use crate::messages::{ResourceDescriptionCidr, ResourceDescriptionDns, ResourceId};
+use ip_network::IpNetwork;
+use proptest::{
+    arbitrary::{any, any_with},
+    strategy::Strategy,
+};
+use std::net::IpAddr;
+
+pub fn dns_resource() -> impl Strategy<Value = ResourceDescriptionDns> {
+    (resource_id(), resource_name(), dns_resource_address())
+        .prop_map(|(id, name, address)| ResourceDescriptionDns { id, address, name })
+}
+
+pub fn cidr_resource() -> impl Strategy<Value = ResourceDescriptionCidr> {
+    (resource_id(), resource_name(), ip_network())
+        .prop_map(|(id, name, address)| ResourceDescriptionCidr { id, address, name })
+}
+
+pub fn resource_id() -> impl Strategy<Value = ResourceId> {
+    any::<u128>().prop_map(ResourceId::from_u128)
+}
+
+pub fn resource_name() -> impl Strategy<Value = String> {
+    any_with::<String>("[a-z]{4,10}".into())
```

Comment:
can we generate any `String`? Since we should support any utf8 string.

Reply:
We can, I just wanted it to be more readable in case of a failed output.

Reply:
A thought occurs https://github.com/kuronpawel/big-list-of-naughty-strings/tree/master